pkg/pgclient/client.go

```diff
@@ -193,3 +201,22 @@ func (c *Client) HealthCheck() error {
 func (c *Client) Queryable() promql.Queryable {
 	return c.queryable
 }
+
+func observeStatementCacheState(x *pgx.Conn) bool {
+	// connections have been opened and are released already
+	// but the Client metrics have not been initialized yet
+	if statementCacheLen == nil || statementCacheCap == nil || statementCacheEnabled == nil {
+		return true
+	}
+	statementCache := x.StatementCache()
+	if statementCache == nil {
+		statementCacheEnabled.Set(0.0)
+		return true
+	}
+
+	statementCacheEnabled.Set(1.0)
+	statementCacheCap.Set(float64(statementCache.Cap()))
+	statementCacheSize := statementCache.Len()
+	statementCacheLen.Observe(float64(statementCacheSize))
```

Comment:
statement cache is per connection, and each observation is the cache size per connection

Reply:
looks great